src/emailmessage.cpp

```diff
@@ -949,3 +1034,29 @@ void EmailMessage::insertInlineImages(const QList<QMailMessagePart::Location> &i
         emit htmlBodyChanged();
     }
 }
+
+const QMailMessagePart* EmailMessage::getCalendarPart()
+{
+    const QMailMessagePart *result = 0;
+    PartFinder finder(QMailMessageContentType("text/calendar"), result);
+    m_msg.foreachPart(finder);
+    return result;
+}
+
+void EmailMessage::saveTempCalendarInvitation(const QMailMessagePart &calendarPart)
+{
+    QString calendarFileName = QStandardPaths::writableLocation(QStandardPaths::TempLocation)
+            + QDir::separator() + calendarPart.identifier();
+
+    QString path = calendarPart.writeBodyTo(calendarFileName);
+    if (!path.isEmpty()) {
+        m_calendarStatus = Saved;
+        m_calendarInvitationUrl = path;
```

Comment:
Does this result in a file:// URI?  If not, maybe it should be converted to one.